File: xServer.D/Controllers/Results/ProfileResult.cs
﻿using System.Collections.Generic;

namespace x42.Controllers.Results
{
    public class ProfileResult
    {
        public string Name { get; set; }

        public string KeyAddress { get; set; }

        public string ReturnAddress { get; set; }

        public string Signature { get; set; }

        public string PriceLockId { get; set; }

        public int Status { get; set; }

        public int ReservationExpirationBlock { get; set; }

        public int BlockConfirmed { get; set; }

        public List<ProfileField> ProfileFields { get; set; }
    }
}

File: xServer.UI/src/app/shared/models/profileresult.ts
export class ProfileResult {
  constructor(name: string, keyAddress: string) {
    this.name = name;
    this.keyAddress = keyAddress;
  }

  public name: string;
  public keyAddress: string;
  public signature: string;
  public priceLockId: string;
  public profileFields: [ProfileField];
}

class ProfileField {
  constructor(value: string, signature: string) {
    this.value = value;
    this.signature = signature;
  }

  public value: string;
  public signature: string;
}
